### ISSUE TITLE
Implement the functional logic of the search bar component

### ISSUE THREAD

**Issue by Uchechukwu-Onye-Igbo** [transport-data/tdc-data-portal#102]
When searching for datasets within the portal, I want to have filtering options and quick access to recent searches,
so that I can easily narrow down search results and quickly revisit my previous searches.

Hint: UI implementation is #29. This implementation should be a component that can be reused on other pages

## Tasklist (TBD - updated by the developer)
* [add tasks]

## Acceptance Criteria
Given that I am on the TDC portal and I want to search for dataset using the search bar (on the homepage, and datasets page),
* [ ] I can see a dropdown with different search filters according to this [UI implementation](https://github.com/transport-data/tdc-data-portal/issues/29) 
* [ ] I can see more filters when i click "show all"
* [ ] I can apply the filters to refine my search
* [ ] I can see my previews searches

**Comment by willy1989cv** [comment]
Documentation: https://github.com/transport-data/tdc-data-portal/tree/main/docs/search-bar-component 
Integration Tests: https://github.com/transport-data/tdc-data-portal/actions/runs/11220721525/job/31189496440#step:14:358

**Comment by Uchechukwu-Onye-Igbo** [comment]
QA Feedback

Hello @willy1989cv please attend to this:

- while search parameters or filters applied using the search bar will produce results, when it arrives on the browse page, the applied filter will is not present in the search bar at the top of the page, the quick filter and the advanced filter.

**Comment by Uchechukwu-Onye-Igbo** [comment]
> QA Feedback
> 
> Hello [@willy1989cv](https://github.com/willy1989cv) please attend to this:
> 
> * while search parameters or filters applied using the search bar will produce results, when it arrives on the browse page, the applied filter will is not present in the search bar at the top of the page, the quick filter and the advanced filter.

this now works. thanks

**Comment by nicolas-becker** [comment]
- The 'in' filter is a bit misleading. The description indicates that it is possible to search for region, countries and cities. In fact, the filter only applied to regions (as described in the docs)
![Image](https://github.com/user-attachments/assets/e6c6f2f8-dda6-4dc0-95ca-f5037a45aa4d)
--> please change text to " filter for regions" or something similar
- Please change to "a selected year" instead of "an year"
- The options listed for mode do not match the actual filter options (e.g. 2W, 3W) 
- To me it would also be the easiest to just drop the explanatory text for now, as the filter options do appear anyways once selected a filter. In this case, please indicate in the docs how/where to include it again if necesary in the future.
- I was wondering if it is possible to include multiple filters in the search query at once. It doesn't seem to be supported currently, but please correct me if I am wrong. Just to fully understand the functionality.

Thanks!

**Comment by willy1989cv** [comment]
Hello @nicolas-becker this implementation is based on the mockups.I will adjust the texts as you described. 
> * I was wondering if it is possible to include multiple filters in the search query at once. It doesn't seem to be supported currently, but please correct me if I am wrong. Just to fully understand the functionality.

Yes, you are right, this search bar is for simple queries. For more advanced search users should use browse page.

Thank you

**Comment by willy1989cv** [comment]
@nicolas-becker I've updated the searchbar as requested, please review again;
>The 'in' filter is a bit misleading. The description indicates that it is possible to search for region, countries and cities. In fact, the filter only applied to regions (as described in the docs)
- [x]  changed text to " filter for regions" or something similar

- [x] Please change to "a selected year" instead of "an year"
>The options listed for mode do not match the actual filter options (e.g. 2W, 3W) : 
- [x]  updated mode filter description to existing options: cars, cycling, heavy rail
>To me it would also be the easiest to just drop the explanatory text for now, as the filter options do appear anyways once selected a filter. In this case, please indicate in the docs how/where to include it again if necesary in the future.
- [x] provided documentation on how to update filters description/options

**Comment by nicolas-becker** [comment]
Perfect, thank you!

**Comment by nicolas-becker** [comment]
I have noticed another strange behaviour related to the asignation of countries, which you can see at the attached video.

I have created a dataset "ab" for test purposes. The country `Libya` was added to the dataset (as you can verify by the metadata displayed in the video). When I search for this data set by its title, it appears. But if I now filter for Libya as a country, the dataset disappears again. 

https://github.com/user-attachments/assets/bf31cd05-8b8b-42e1-9e5f-597c20c6542a

**Comment by Gutts-n** [comment]
Hello @nicolas-becker, I didn't find your dataset, did you delete it?

**Comment by Gutts-n** [comment]
Hey @nicolas-becker this is happening because the dataset that you are trying to filter is private. Do you think we should remove private datasets from this page to prevent this from happening to other users?

**Comment by nicolas-becker** [comment]
Hi @Gutts-n, yes, that should not happen to a user. But I am afraid I didn't understand the problem entirely. 

I was changing the dataset to public and noticed that I am getting a similar issue when displaying "my datasets" in the dashboard. The function seems to be unstable as the dataset keeps appearing and disappearing.

**Comment by nicolas-becker** [comment]
I tried to capture the behaviour in a screencast:

- I have three datasets in my profile currently (`ab`, `abc`, `Test Data NB`).
- `ab` disappeared from the listing when I changed it to public. 
- When loading my datasets it first only displays the private `Test Data NB` 
- After a few seconds or after reloading the page, `Test Data NB` is disappears and `abc` (draft) appears instead
https://github.com/user-attachments/assets/c5cc518e-bc19-41ae-9825-abd97b2fa695

@Gutts-n

**Comment by Gutts-n** [comment]
@nicolas-becker take a look now and give me feedback.

**Comment by nicolas-becker** [comment]
@Gutts-n Hi Leonardo, thanks for the fix and for the explanation. Seems to be working now. Could you please add this hint to the Title/Slug section of the dataset creation process? Currently the user only gets an error message if the title is less than 2 characters long. 

Also, I was wondering if it will be possible to search for datasets of specific countries in the search bar:
> Hi @willy1989cv , sorry to come back to this issue again. I was trying to search for a dataset for a specific country, (e.g. India, which has datasets already on the platform). When I type the country name in the search bar, the dataset related to India will not appear. This means that the free search functionality is limited to the data set name, right? I think we already had a conversation on this previously, but I would need a confirmation. Thank you!

**Comment by Gutts-n** [comment]
Hey @nicolas-becker the country problem is related with the dataset with 2 characters, if you create a dataset with a name with more than 4 characters the filter should work correctly.

**Comment by willy1989cv** [comment]
>When I type the country name in the search bar, the dataset related to India will not appear. This means that the free search functionality is limited to the data set name, right?

@nicolas-becker yes, the text search only searches for title and description field.

**Comment by nicolas-becker** [comment]
Thanks! 

@Gutts-n Please let me know when you have fixed this and added the respective hint/error message to the user in the dataset creation process.

@willy1989cv Okay, so the only way for the user to search for a dataset specific for a country would be via the geography page right?

**Comment by willy1989cv** [comment]
>so the only way for the user to search for a dataset specific for a country would be via the geography page right?

@nicolas-becker users can search for a specific country in search page
![image](https://github.com/user-attachments/assets/e23b1d09-8e3f-4e54-838d-4de57305469e)

**Comment by nicolas-becker** [comment]
@willy1989cv 

I see, thank you. In the future we will need to add the `Country` filter to the search bar, so the user can directly search for their country of interest:
![image](https://github.com/user-attachments/assets/3d35b7c8-087a-45b7-a514-2c5351ceec19)

This is currently not critical, but will become important once the platform contains country-specific data. Probably we'll also need to add a `Country` filter in the quick filters or replace the `Region` filter with the `Location` filter. 

Will we be able to (more or less) easily fix these kinds of issues on our own? Maybe you can leave a hint where in the code we find these functionalities. Thanks!

**Comment by Gutts-n** [comment]
Hey @nicolas-becker, I pushed a fix to Solr. Could you take a look at the problem now and give me feedback? I also added a hint under the title and name on the dataset create/edit pages to inform about the min of 2 characters.

**Comment by nicolas-becker** [comment]
Hi @Gutts-n thanks for the fix! I just had a look, the issue on the dataset browsing page seems to be fixed now.

> the country problem is related with the dataset with 2 characters, if you create a dataset with a name with more than 4 characters the filter should work correctly.

Just want to make sure whether the minimum of 2 chars is correct now since the latest fix?

<img width="419" alt="image" src="https://github.com/user-attachments/assets/a81d4f6c-defa-4c35-8ec2-fc5e8e88d100">

I see that you changed the name from Slug back to Name. I still think that asking for both title and name is confusing for the user. Would it be necessary for the user to edit the name, as it is automatically filled with the information from the title? As an alternative: remove the input field of the Name/Slug and replace the hint of "min. 2 characters" in the Title field with the newly created URL of the dataset, accoring to the automatically generated name.  The error message stating that the field should contain at least 2 characters is enough (just as it is currently implemented).

Another thing: The cross from the search field seems to overlap the dropdown menu:
![image](https://github.com/user-attachments/assets/95d2c571-3aae-4dc4-9f84-98768da500f0)

**Comment by willy1989cv** [comment]
we can discuss this tomorrow @nicolas-becker

**Comment by nicolas-becker** [comment]
Alrught @willy1989cv thank you! And just to be clear, I am not necessarily asking you to do any changes, because I know we already made a decision on this. I would just like to know if we will be able to make these kinds of changes ourselves and, if yes, how.

**Comment by willy1989cv** [comment]
>Yes exactly, adding a country filter there.

 I will add documentation on how to add a new filter to the search component @nicolas-becker

**Comment by nicolas-becker** [comment]
Thanks @willy1989cv

**Comment by willy1989cv** [comment]
@nicolas-becker the issue is in peer-review, as soon as it is merged I'll let you know.
changes:
- added country search in the searchbar
- updated search-bar-component documentation

**Comment by nicolas-becker** [comment]
Very nice, thank you @willy1989cv
